classes/core/PKPRouter.php

```diff
@@ -357,20 +231,15 @@ public function getIndexUrl($request)
      */
     public function getCacheFilename($request)
     {
-        // must be implemented by sub-classes
-        assert(false);
+        throw new Exception('Unimplemented');
```

Comment:
I forget this case in particular, but in some cases changes from `assert(false)` to `abstract function` caused a need to refactor the tests. I did some of this but drew a line on the rest. This is a slight improvement, and it would be a good idea to move this to an `abstract function` later.

Reply:
Actually, I think I do remember this: Some subclasses legitimately do not implement the function, as they have no caching implemented.